### DISCUSSION TITLE
CreatedBy in Email

### DISCUSSION THREAD

**Discussion by kasgithub8** [Budibase/budibase#6054] [Help] [answered]
Hey,

is there any way to get the "CreatedBy" Email into the Email field [Send Email (SMTP)] for Automations?

Thanks

**Comment by melohagan** [comment, marked as answer]
Hey @kasgithub8 

I'm not sure what you mean by the "CreatedBy" email. Do you mean the user that triggered the automation?

**Reply by kasgithub8** [reply]
Hey @melohagan,

Sorry! For example, somebody creates a contract, product etc.. The Auto-Column "CreatedBy" should store the Email of the creator
I want to have that E-Mail in the automation. Is that possible?

**Reply by melohagan** [reply]
Thanks @kasgithub8 I understand now.

Assuming you have a **Row Created** automation action, then in your automations steps you could access the following binding:

`{{ trigger.row[Created By].0.email }}`

**Reply by kasgithub8** [reply]
Hey @melohagan, 

sorry I just noticed that you answered. I have a Row Updated automation. 
Putting "{{ trigger.row[Created By].0.email }}" in the "Send To" Field does not work. :(

**Reply by melohagan** [reply]
Hey @kasgithub8 

It works for me in a published app. 
Maybe you found is that it didn't work in your testing? That would be because when you create a row in your test, there is no user associated to the creation of test data.

![Screenshot 2022-06-07 at 15 29 36](https://user-images.githubusercontent.com/101575380/172406150-0360ef02-c519-46fc-abb5-6f70bae30e3e.png)

**Reply by kasgithub8** [reply]
Hi @melohagan,

omg 🤦🏽‍♂️ . I just remembered that I had to delete and re-add the "Created By" field because it bugs out sometimes. When I added it again, I named it "CreatedBy" so {{ trigger.row[Created By].0.email }} obviously wouldn't work. {{ trigger.row[CreatedBy].0.email }} works perfectly now. 
Thank you. 😄

**Reply by melohagan** [reply]
Perfect! As there is no space in that column name, you should also be able to do `{{ trigger.row.CreatedBy.0.email }}` but both are grand!

**Reply by kasgithub8** [reply]
One more thing @melohagan,

is there a way to put a link to the /:id of the updated/created row in the email? 
I can't seem to find a way to make it dynamic.

Thanks!

**Reply by melohagan** [reply]
Hi @kasgithub8 

Yes, it should simply be the `{{ trigger.id }}` binding, if you are using the **Row Created/Updated** trigger. 

If you're using the internal table, this id will be a random string of letters though - may be better to pick out a different column to be more readable.

**Reply by kasgithub8** [reply]
So @melohagan,

`{{ trigger.id }}` gives me 

> ro_ta_d1a6973603634cd29fa3a978705e309d_34f79ff003bf4f689317fa11cf8fc412 

The link I would need would be something like 

> https://.../app/contracttest#/contract/60

The 60 is my "Auto ID". Can I like combine the link with a `{{ trigger.row }} ` item?

**Reply by melohagan** [reply]
The `{{ trigger.row[Auto ID] }}` binding worked for me

**Reply by kasgithub8** [reply]
@melohagan you're a blessing. Thank you so much!
